bitcoin/src/crypto/sighash.rs

```diff
@@ -94,8 +94,8 @@ hashes::impl_serde_for_newtype!(TapSighash);
 impl_message_from_hash!(TapSighash);
 
 /// Efficiently calculates signature hash message for legacy, SegWit and Taproot inputs.
-#[derive(Debug)]
-pub struct SighashCache<T: Borrow<Transaction>> {
+#[derive(Debug, Clone)]
+pub struct SighashCache<T: Borrow<Transaction> + Clone> {
```

Comment:
That additional `Clone` bound shouldn't be needed and it's harmful. The derive macro inserts its own and forcing it prevents `&mut Transaction` from being used which specifically goes against the intent of `BorrowMut`.